code/parachain/frame/assets/src/lib.rs

```diff
@@ -309,6 +309,73 @@ pub mod pallet {
 			)?;
 			Ok(().into())
 		}
+
+		/// Transfer all free balance of the `asset` from `source` to `dest`.
+		///
+		/// # Errors
+		///  - When `origin` is not signed.
+		///  - If the `dest` cannot be looked up.
+		#[pallet::weight(T::WeightInfo::force_transfer_all())]
+		#[pallet::call_index(10)]
+		pub fn force_transfer_all(
+			origin: OriginFor<T>,
+			asset: T::AssetId,
+			source: <T::Lookup as StaticLookup>::Source,
+			dest: <T::Lookup as StaticLookup>::Source,
+			keep_alive: bool,
+		) -> DispatchResult {
+			ensure_root(origin)?;
```

Comment:
@kkast is it really should be ensure_root or maybe some gov pallet?

but overall everything is good.

Reply:
we currently use root for force transfer, but in theory we can change it.